src/string.cr

```diff
@@ -3388,6 +3393,11 @@ class String
     match_result.try &.begin
   end
 
+  # Same as `rindex` but raises NilAssertionError if no index found
+  def rindex!(search, offset = size - 1) : Int32
```

Comment:
That's incorrect, it should have different overloads based on `search` type (`String | Char | Regex`).